transport/core/config.go

```diff
@@ -0,0 +1,45 @@
+package core
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+)
+
+// TLSClientAuthClientConfig returns a new client authentication TLS
+// configuration that can be used for a client using client auth
+// connecting to the named host.
+func TLSClientAuthClientConfig(cert tls.Certificate, host string) *tls.Config {
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      SystemRoots,
```

Comment:
Again, it would be nice to be able to replace these system roots with a short list comprising of CA providers that are known to be trusted.